File: src/DeskFlow.Core/Entities/Gamificacion.cs
using DeskFlow.Core.Common;

namespace DeskFlow.Core.Entities;

public class Logro
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public string Clave { get; set; } = string.Empty;
    public string Nombre { get; set; } = string.Empty;
    public string Descripcion { get; set; } = string.Empty;
    public string Icono { get; set; } = "🏆";
    public string Criterio { get; set; } = string.Empty;
    public int PuntosRecompensa { get; set; }
    public int Orden { get; set; }
}

public class LogroAgente : TenantEntity
{
    public Guid UsuarioId { get; set; }
    public Guid LogroId { get; set; }
    public DateTime FechaObtenido { get; set; } = DateTime.UtcNow;

    public Tenant Tenant { get; set; } = null!;
    public Usuario Usuario { get; set; } = null!;
    public Logro Logro { get; set; } = null!;
}

public class DashboardPersonalizado : TenantEntity
{
    public Guid UsuarioId { get; set; }
    public string WidgetsJson { get; set; } = "[]";
    public DateTime FechaModif { get; set; } = DateTime.UtcNow;

    public Tenant Tenant { get; set; } = null!;
    public Usuario Usuario { get; set; } = null!;
}

public class ReporteCompartido : TenantEntity
{
    public string Token { get; set; } = string.Empty;
    public string Titulo { get; set; } = string.Empty;
    public string ConfigJson { get; set; } = "{}";
    public string? DatosJson { get; set; }
    public Guid? CreadoPorId { get; set; }
    public DateTime FechaCreacion { get; set; } = DateTime.UtcNow;
    public DateTime? FechaExpiracion { get; set; }

    public Tenant Tenant { get; set; } = null!;
    public Usuario? CreadoPor { get; set; }
}

File: src/DeskFlow.Web/src/api/gamificacion.ts
import apiClient from './client';

export interface LogroResumen {
  icono: string;
  nombre: string;
}

export interface RankingItem {
  posicion: number;
  usuarioId: string;
  nombreCompleto: string;
  rol: string;
  ticketsResueltos: number;
  ticketsEnSLA: number;
  porcentajeSLA: number;
  promedioCsat: number | null;
  tiempoPromedioHoras: number;
  puntos: number;
  logros: LogroResumen[];
}

export interface LogroDto {
  id: string;
  clave: string;
  nombre: string;
  descripcion: string;
  icono: string;
  criterio: string;
  puntosRecompensa: number;
  obtenido: boolean;
  fechaObtenido: string | null;
}

export interface PerfilGamificacion {
  usuarioId: string;
  nombreCompleto: string;
  posicion: number;
  puntos: number;
  ticketsResueltos: number;
  porcentajeSLA: number;
  promedioCsat: number | null;
  logros: LogroDto[];
}

export interface WidgetConfig {
  id: string;
  tipo: string;
  titulo: string;
  visible: boolean;
  orden: number;
}

export interface ReporteCompartido {
  id: string;
  token: string;
  titulo: string;
  url: string;
  fechaCreacion: string;
  fechaExpiracion: string | null;
}

export interface CrearReporteDto {
  titulo: string;
  configJson: string;
  datosJson: string;
  diasExpiracion?: number;
}

export const gamificacionApi = {
  getRanking: (periodo: 'semana' | 'mes' | 'total' = 'mes') =>
    apiClient.get<RankingItem[]>('/gamificacion/ranking', { params: { periodo } }).then(r => r.data),

  getMiPerfil: () =>
    apiClient.get<PerfilGamificacion>('/gamificacion/mi-perfil').then(r => r.data),

  getMisLogros: () =>
    apiClient.get<LogroDto[]>('/gamificacion/mis-logros').then(r => r.data),

  checkBadges: (usuarioId: string) =>
    apiClient.post('/gamificacion/check-badges', null, { params: { usuarioId } }),

  getMisWidgets: () =>
    apiClient.get<WidgetConfig[]>('/gamificacion/mis-widgets').then(r => r.data),

  saveWidgets: (widgets: WidgetConfig[]) =>
    apiClient.put('/gamificacion/mis-widgets', widgets),

  getMisReportes: () =>
    apiClient.get<ReporteCompartido[]>('/gamificacion/reportes').then(r => r.data),

  crearReporte: (dto: CrearReporteDto) =>
    apiClient.post<ReporteCompartido>('/gamificacion/reportes', dto).then(r => r.data),

  eliminarReporte: (id: string) =>
    apiClient.delete(`/gamificacion/reportes/${id}`),
};
